staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/types.go

```diff
@@ -174,6 +174,9 @@ type CustomResourceDefinitionVersion struct {
 	// Top-level and per-version columns are mutually exclusive.
 	// Per-version columns may not all be set to identical values (top-level columns should be used instead)
 	// This field is alpha-level and is only honored by servers that enable the CustomResourceWebhookConversion feature.
+	// NOTE: CRDs created prior to 1.13 populated the top-level additionalPrinterColumns field by default. To apply an
```

Comment:
The "Defaults to a created-at column." is not really correct, is it? It only defaults if at least one version has additionalPrinterColumns. Otherwise, the global field is defaulted.

Reply:
> It only defaults if at least one version has additionalPrinterColumns. Otherwise, the global field is defaulted

That was the previous approach. @mbohlool [pointed out](https://github.com/kubernetes/kubernetes/pull/70211#discussion_r230518621) that in this way, we have the problem that if a user first created a CRD without specifying global/per-version columns, and then tries to patch per-version columns, the patch request will fail the mutual-exclusive validation.

We decided to drop the validation and only do logical default to improve the user experience for new CRDs. For CRDs created prior to 1.13, we cannot avoid the issue so we document in the API and in release note. The change is in 6th commit.

Reply:
Found it. Is it written in the release notes? Is it a breaking API change to change (remove) defaults? Strictly speaking I think it is. /cc @liggitt

Reply:
> Is it a breaking API change to change (remove) defaults? Strictly speaking I think it is.

Making the existing defaulted field mutually exclusive with the per-version fields (which is required if we want to only have per-version fields in v1) unavoidably requires changing the defaulting behavior.

I can't think of any scenarios this change causes problems with:
* existing persisted CRDs retain current behavior (with respect to what columns are returned by server-side printing)
* existing CRD manifests retain current behavior when submitted to a new API server (with respect to what columns are returned by server-side printing)
* skewed HA API servers across a 1.12/1.13 boundary would serve a CRDs without this field identically
* the field is marked as optional in the openapi schema, so generated clients will tolerate it being unset

I'm comfortable proceeding as is